### PR TITLE
Set series to dataframe

### PR DESCRIPTION
**Pull request by wuisawesome**
## What do these changes do?
In QueryCompiler's setitem we check if a list-like value has columns in which case we want to set the first column instead of calling the list constructor. 

This doesn't fully match pandas behavior on all 2d objects like numpy arrays but it should ensure that pandas types are handled correctly. 

## Related issue number
#541 

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/dataframe.py [hunk 1]
```diff
@@ -4754,12 +4754,22 @@ def __setitem__(self, key, value):
         if not isinstance(key, str):
 
             def setitem_without_string_columns(df):
+                df = df.copy(True)
```
Comment (devin-petersohn) [inline modin/pandas/dataframe.py]: We probably should document why this is needed.

File: modin/pandas/test/test_dataframe.py [hunk 2]
```diff
@@ -4460,6 +4461,25 @@ def test___setitem__(self, data):
...
+        modin_df[modin_df.columns[-1]] = pd.DataFrame(modin_df[modin_df.columns[0]])
+        pandas_df[pandas_df.columns[-1]] = pandas.DataFrame(
+            pandas_df[pandas_df.columns[0]]
+        )
+        df_equals(modin_df, pandas_df)
+
+        modin_df = pd.DataFrame(data)
+        pandas_df = pandas.DataFrame(data)
+
+        # TODO: These commented out lines are some nasty edge cases we don't support
```
Comment (devin-petersohn) [inline modin/pandas/test/test_dataframe.py]: What is stopping us from supporting these edge cases?

Comment (wuisawesome) [reply modin/pandas/test/test_dataframe.py]: We could if statement this out, but it's unclear/undocumented when shapes should be coerced. I'm mostly not sure if there's an underlying reason why there should be different behavior between setting a nonexistent column vs. one that exists

File: modin/backends/pandas/query_compiler.py [hunk 3]
```diff
@@ -1983,6 +1983,7 @@ def setitem(self, axis, key, value):
         """
 
         def setitem(df, internal_indices=[]):
+            df = df.copy()
```
Comment (williamma12) [inline modin/backends/pandas/query_compiler.py]: Let's add a try-except block similar to #569 to avoid making extraneous copies?

**Comment by williamma12** [comment]
This should fix #525

File: modin/backends/pandas/query_compiler.py [hunk 4]
```diff
@@ -1983,16 +1983,31 @@ def setitem(self, axis, key, value):
         """
 
         def setitem(df, internal_indices=[]):
-            if len(internal_indices) == 1:
-                if axis == 0:
-                    df[df.columns[internal_indices[0]]] = value
+            try:
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py]: instead of duplicating this code can we create a subroutine that gets called?

File: modin/backends/pandas/query_compiler.py [hunk 5]
```diff
@@ -1983,16 +1983,31 @@ def setitem(self, axis, key, value):
...
                 else:
-                    df.iloc[internal_indices[0]] = value
-            else:
-                if axis == 0:
-                    df[df.columns[internal_indices]] = value
+                    if axis == 0:
+                        df[df.columns[internal_indices]] = value
+                    else:
+                        df.iloc[internal_indices] = value
+            except Exception:
```
Comment (devin-petersohn) [inline modin/backends/pandas/query_compiler.py]: What type of exception gets thrown here?

**Comment by devin-petersohn** [comment]
The uncovered lines are because codecov does not know when Ray runs some code. Ok to merge.
